### PR TITLE
dbeaver/pro#1647 Deep copy task configuration when copying task

### PR DESCRIPTION
**Pull request by ShadelessFox**

### CODE REVIEW

**Review by E1izabeth — APPROVED** [review]

File: bundles/org.jkiss.utils/src/org/jkiss/utils/CommonUtils.java [hunk 1]
```diff
@@ -228,6 +231,47 @@ public static <T> List<T> copyList(@Nullable Collection<T> theList) {
         }
     }
 
+    @NotNull
+    public static <T> List<T> deepCopy(@NotNull List<T> src) {
```
Comment (LonwoLonwo) [inline bundles/org.jkiss.utils/src/org/jkiss/utils/CommonUtils.java]: I think a "Missing a Javadoc comment." has some meaning here.

Comment (ShadelessFox) [reply bundles/org.jkiss.utils/src/org/jkiss/utils/CommonUtils.java]: I believe the method name is already self-explanatory.

**Review by LonwoLonwo — APPROVED** [review]

**Comment by MashaKorax** [comment]
verified

File: bundles/org.jkiss.utils/src/org/jkiss/utils/CommonUtils.java [hunk 2]
```diff
@@ -228,6 +231,47 @@ public static <T> List<T> copyList(@Nullable Collection<T> theList) {
         }
     }
 
+    @NotNull
```
Comment (serge-rider) [inline bundles/org.jkiss.utils/src/org/jkiss/utils/CommonUtils.java]: It is better to move these functions to BeanUtils

File: bundles/org.jkiss.utils/src/org/jkiss/utils/CommonUtils.java [hunk 3]
```diff
@@ -228,6 +231,47 @@ public static <T> List<T> copyList(@Nullable Collection<T> theList) {
...
+        final List<T> dst = tryInstantiateOrDefault(src.getClass(), ArrayList::new);
+        for (T element : src) {
+            dst.add(deepCopy(element));
+        }
+        return dst;
+    }
+
+    @NotNull
+    public static <K, V> Map<K, V> deepCopy(@NotNull Map<K, V> src) {
+        final Map<K, V> dst = tryInstantiateOrDefault(src.getClass(), HashMap::new);
```
Comment (serge-rider) [inline bundles/org.jkiss.utils/src/org/jkiss/utils/CommonUtils.java]: I don't like HashMap here, even as default. LinkedHashMap<> is better.

**Review by serge-rider — CHANGES_REQUESTED** [review]

**Comment by MashaKorax** [comment]
verified

**Review by serge-rider — APPROVED** [review]

**Pull request merged**